litestar/handlers/base.py

```diff
@@ -254,6 +254,16 @@ def handler_name(self) -> str:
         """
         return get_name(unwrap_partial(self.fn))
 
+    @property
+    def handler_fullname(self) -> str:
+        """Get the qualified name of the handler function.
+
+        Returns:
+            Qualified name of the handler function
+        """
+        handler_fn = unwrap_partial(self.fn)
+        return handler_fn.__module__ + "." + handler_fn.__qualname__
```

Comment:
I think the logic here should be 

```suggestion
        return self.name or self.handler_name
```
so custom handler names show up as well, which means the handler name shown in the exception matches e.g. the one you'd put into `route_reverse`.

Reply:
But this way we're loosing package and class name...

Reply:
How about just using `str(handler.fn)`? I'm hesitant to add yet another handler identifier that does something slightly different; We already have `handler_id`, `name` and `handler_name`.

Maybe we could change `handler_name` in v3 to use the fully qualified name instead.

Reply:
Actually `str(self)` does exactly that (module+qualname)